utils/s2n_map.c

```diff
@@ -160,6 +160,20 @@ int s2n_map_add(struct s2n_map *map, struct s2n_blob *key, struct s2n_blob *valu
     return 0;
 }
 
+int s2n_map_offer(struct s2n_map *map, struct s2n_blob *key, struct s2n_blob *value)
```

Comment:
Should we just have an s2n_map_put() that does add-or-replace insertion?  The offer() way isn't something I've ever seen before, so it took me a bit to understand what it's doing and why. It favors leaving the former value if it's there, but any reason to favor that?